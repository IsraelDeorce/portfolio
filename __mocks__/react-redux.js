const mock = jest.createMockFromModule('react-redux')

const dispatch = jest.fn((obj) => obj)
dispatch.session = {
  update: jest.fn(),
  changeLanguage: jest.fn(),
  switchTheme: jest.fn()
}
const selectors = {
  session: {
    language: 'en',
    theme: 'dark',
    isSideBarOpen: false
  }
}

mock.useDispatch = jest.fn(() => dispatch)
mock.useSelector = jest.fn((fn) => fn(selectors))

module.exports = mock
