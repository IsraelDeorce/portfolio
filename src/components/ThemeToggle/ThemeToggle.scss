.toggle .react-toggle-track {
  background-color: #4D4D4D !important;
  & > * {
    display: flex;
    align-items: center;
    color: #FFCC33;
  }
}
