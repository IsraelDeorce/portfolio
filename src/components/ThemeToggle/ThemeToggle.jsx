import { useDispatch } from 'react-redux'
import { FaMoon, FaSun } from 'react-icons/fa'
import Toggle from 'react-toggle'
import 'react-toggle/style.css'

import './ThemeToggle.scss'

function ThemeToggle() {
  const dispatch = useDispatch()

  return (
    <Toggle
      onChange={dispatch.session.switchTheme}
      className='toggle'
      icons={{
        checked: <FaSun />,
        unchecked: <FaMoon />
      }}
    />
  )
}

export default ThemeToggle
