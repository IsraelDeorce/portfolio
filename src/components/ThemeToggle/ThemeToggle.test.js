import { useDispatch } from 'react-redux'
import { FaMoon, FaSun } from 'react-icons/fa'
import ThemeToggle from './ThemeToggle'

describe('<ThemeToggle/>', () => {
  const dispatch = useDispatch()

  afterEach(() => jest.clearAllMocks())

  it('renders a Toggle component from react-togle with a sun and moon as icons', () => {
    const expected = {
      checked: <FaSun />,
      unchecked: <FaMoon />
    }
    expect(shallow(<ThemeToggle />).find('Toggle').prop('icons')).toStrictEqual(expected)
  })

  it('calls the switch theme function when the user clicks on the Toggle', () => {
    shallow(<ThemeToggle />).find('Toggle').simulate('change')
    expect(dispatch.session.switchTheme).toHaveBeenCalled()
  })

  describe('Snapshot', () => {
    it('renders without crashing using shallow', () => {
      expect(shallow(<ThemeToggle />)).toMatchSnapshot()
    })

    it('renders without crashing using render', () => {
      expect(render(<ThemeToggle />)).toMatchSnapshot()
    })

    it('renders without crashing using mount', () => {
      expect(mount(<ThemeToggle />)).toMatchSnapshot()
    })
  })
})
