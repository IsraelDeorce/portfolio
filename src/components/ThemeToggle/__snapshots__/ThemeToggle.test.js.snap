// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeToggle/> Snapshot renders without crashing using shallow 1`] = `
<Toggle
  className="toggle"
  icons={
    Object {
      "checked": <FaSun />,
      "unchecked": <FaMoon />,
    }
  }
  onChange={[MockFunction]}
/>
`;
